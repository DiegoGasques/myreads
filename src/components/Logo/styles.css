.logo h1 {
  margin: 0;
  color: #fff;
}

.logo h1.sm {
  font-size: 1rem;
}

.logo h1.md {
  font-size: 1.7rem;
}

.logo h1.lg {
  font-size: 3rem;
}
