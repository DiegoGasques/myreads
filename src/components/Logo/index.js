import React from "react";
import { Link } from "react-router-dom";
import classNames from "classnames";
import PropTypes from "prop-types";

import "./styles.css";

function Logo({ text, className, path, size }) {
  return (
    <Link to={path}>
      <div className={classNames("logo", className)}>
        <h1 className={classNames({ md: !size })}>{text}</h1>
      </div>
    </Link>
  );
}

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  image: PropTypes.string,
  className: PropTypes.string,
  path: PropTypes.string.isRequired,
  size: PropTypes.string
};

export default Logo;
