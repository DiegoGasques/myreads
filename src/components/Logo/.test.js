import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Logo from "./index";

it("Should match Logo component snapshot without optional props", () => {
  const wrapper = shallow(<Logo text="Logo" path="/" />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should match Logo component snapshot with optional props", () => {
  const wrapper = shallow(
    <Logo text="Logo" path="/" className="great-logo" size="sm" />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
