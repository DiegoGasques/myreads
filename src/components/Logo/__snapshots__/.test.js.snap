// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Logo component snapshot with optional props 1`] = `
<Link
  replace={false}
  to="/"
>
  <div
    className="logo great-logo"
  >
    <h1
      className="sm"
    >
      Logo
    </h1>
  </div>
</Link>
`;

exports[`Should match Logo component snapshot without optional props 1`] = `
<Link
  replace={false}
  to="/"
>
  <div
    className="logo"
  >
    <h1
      className="md"
    >
      Logo
    </h1>
  </div>
</Link>
`;
